addons/mail/static/src/models/file_uploader.js

```diff
@@ -126,7 +126,7 @@ registerModel({
         async _performUpload({ files }) {
             const composer = this.composerView && this.composerView.composer; // save before async
             const thread = this.thread; // save before async
-            const chatter = this.chatterOwner; // save before async
+            const chatter = this.chatterOwner || this.attachmentBoxView && this.attachmentBoxView.chatter; // save before async
```

Comment:
this wont lead into bug if `this.chatterOwner = true` and `this.attachmentBoxView = null`? and `this.attachmentBoxView.chatter` will report a problem i suppose?

Reply:
when `this.chatterOwner` has a value it doesn't execute the rest `||` is immediately returning the value

Reply:
ah, make sense!

Reply:
```javascript
this.chatterOwner || (this.attachmentBoxView && this.attachmentBoxView.chatter);
```
This would make the logic more obvious ?

Reply:
logically speaking, indeed.